ffi/rtc.lua

```diff
@@ -0,0 +1,210 @@
+--[[--
+Module for interfacing with the RTC (real time clock).
+
+This module provides the ability to schedule wakeups through RTC.
+
+See <http://man7.org/linux/man-pages/man4/rtc.4.html> for technical details.
+
+@module ffi.rtc
+]]
+
+local ffi = require("ffi")
+local bor = bit.bor
+local C = ffi.C
+
+-- Load header definitions for functions and constants to the ffi.C namespace.
+local dummy = require("ffi/posix_h")
+local dummy = require("ffi/rtc_h")
+
+-----------------------------------------------------------------
+
+local RTC = {
+    _wakeup_scheduled = false,   -- Flipped in @{setWakeupAlarm} and @{unsetWakeupAlarm}.
+    _wakeup_scheduled_ptm = nil, -- Stores a reference to the time of the last scheduled wakeup alarm.
+}
+
+--[[--
+Adds seconds to epoch.
+
+@int seconds_from_now Number of seconds.
+@treturn int (cdata) Epoch.
+--]]
+function RTC:secondsFromNowToEpoch(seconds_from_now)
+    local t = ffi.new("time_t[1]")
+    t[0] = C.time(nil)
+    t[0] = t[0] + seconds_from_now
+    local epoch = C.mktime(C.localtime(t))
+    return epoch
+end
+
+--[[--
+Set wakeup alarm.
+
+If you want to set the alarm to a certain amount of time from now,
+you can process your value with @{secondsFromNowToEpoch}.
+
+@int Epoch.
+@enabled bool Whether the call enables or disables the alarm. Defaults to true.
+
+@treturn bool Success.
+@treturn re Error code (if any).
+@treturn err Error string (if any).
+--]]
+function RTC:setWakeupAlarm(epoch, enabled)
+    enabled = (enabled ~= nil) and enabled or true
+
+    local ptm = C.gmtime(ffi.new("int[1]", epoch))
```

Comment:
@NiLuJe This was my new simplified approach (without a whole time = time_t etc.) but apparently I accidentally made the module ARM-specific because on my computer it errors saying it should be uint_64. :man_facepalming: 

Anyway, not an issue for the initial Kobo implementation but I figured I'd leave a note. I'll (hopefully) look into recomplexifying it after the finishing touches on the frontend implementation. Which I expected to have finished already but I had to rush that MuPDF patch and stuff. :-P